src/base/broadcast.jl

```diff
@@ -93,28 +93,29 @@ broadcast_rep_child(r::SparseData) = r.lvl
 broadcast_rep_child(r::DenseData) = r.lvl
 broadcast_rep_child(r::RepeatData) = r.lvl
 
-broadcast_rep(::BroadcastRepDenseStyle, f, args) = DenseData(broadcast_rep(f, map(broadcast_rep_child, args)))
+broadcast_rep(::BroadcastRepDenseStyle, f, args) = DenseData(broadcast_rep(f, map(broadcast_rep_child, args)), Int)
```

Comment:
is there ever a case where `SparseList{Int16} + SparseList{Int16} = SparseList{Int16}`, or will it always be Int? I suppose it doesn't make sense to propagate position types?

Reply:
I am not sure.  I think it does make sense to do so (the right thing to do probably eventually ends with calling the type promotion rules), but I am not sure it will ever cause any problems.

Though on the other hand how does broadcast interact with continuous indicies (@nullplay )?

Reply:
I suppose this is also handled through size inference in the broadcast as well.

Reply:
I think for now we should leave broadcast edge stuff until later.